llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -2681,6 +2681,10 @@ bool SIInsertWaitcnts::run(MachineFunction &MF) {
         if (CT == LOAD_CNT || CT == DS_CNT || CT == STORE_CNT)
           continue;
 
+        if (!ST->hasImageInsts() &&
+            (CT == EXP_CNT || CT == SAMPLE_CNT || CT == BVH_CNT))
```

Comment:
Using hasImageInsts to control EXP_CNT looks weird - will this be changed again later?

Reply:
We may add a feature, but the problem with it we do not like negative features. So a FeatureEXPInsts will need to be added almost everywhere. Sound very invasive.